SU2_CFD/src/output/COutput.cpp

```diff
@@ -389,6 +389,9 @@ void COutput::WriteToFile(CConfig *config, CGeometry *geometry, OUTPUT_TYPE form
   /*--- File writer that will later be used to write the file to disk. Created below in the "switch" ---*/
   CFileWriter *fileWriter = nullptr;
 
+  /*--- Set current time iter even if history file is not written ---*/
+  curTimeIter = config->GetTimeIter();
```

Comment:
If curTimeIter is a member variable, why are you having to set its value here?
It was already used in line 418/421.
So do we have the same problem with innerIter and outerIter? Can you fix this closer to the current place where these iteration counters are set? (history file judging from the comment)

Reply:
The problem only arises when I set the inner iterations to be 0 in an unsteady dual time-stepping simulation. I guess that, since in this case the history file is not updated, the curTimeIter is not updated and remains 0 when writing the output files. In all other cases where the history file is written it is not necessary. I guess I could use an if-statement for that but I don't think that it is necessary. I don't know if the problem persist with the outer iter, I will check (I guess it only matters during a steady simulation when (WRT_*_OVERWRITE is set to YES in the config file).

Reply:
Yes but please look for a single place where to put setting these variables, potentially moving where they are set at the moment.

Reply:
Sure, I missed that request before. I have created a setCurTimeIter function in the COutput class and I call it when the Solve function is called. I have checked also for multi-zone problems and it works fine.